stock_valuation_fifo_lot/models/product.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2022 Ecosoft Co., Ltd (https://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+from odoo.tools import float_is_zero
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def _get_all_candidates(self, company):
```

Comment:
IMHO, to increase reusability, we can refactor the `sorted` operation...
```suggestion
    def _get_all_candidates(self, company, sort_by=None):
```